### PR TITLE
--download: Use `time.monotonic()` and rework code to prevent `ZeroDivisionError` specific handling

### PR DESCRIPTION
**Pull request by BoboTiG**
It simplifies speed/percentage handling. It also expands coverage as testing previous `ZeroDivisionError` corner-cases is not easy.

### CODE REVIEW

File: httpie/downloads.py [hunk 1]
```diff
@@ -22,6 +22,8 @@
 
 PARTIAL_CONTENT = 206
 
+TIME_FUNC = monotonic
```
Comment (jakubroztocil) [inline httpie/downloads.py:25]: Why the constant?

**Review by jakubroztocil — CHANGES_REQUESTED** [review]
Using `monotonic` is a good idea. However, it doesn’t remove the need for the handling of `ZeroDivisionError`.

File: httpie/downloads.py [hunk 2]
```diff
@@ -407,26 +409,22 @@ def run(self):
...
 
         if now - self._prev_time >= self._update_interval:
             downloaded = self.status.downloaded
-            try:
-                speed = ((downloaded - self._prev_bytes)
-                         / (now - self._prev_time))
-            except ZeroDivisionError:
-                speed = 0
+            speed = ((downloaded - self._prev_bytes)
+                     / (now - self._prev_time))
```
Comment (jakubroztocil) [inline httpie/downloads.py:417]: `downloaded - self._prev_bytes` can be `0` if nothing new has been downloaded

Comment (BoboTiG) [reply httpie/downloads.py:417]: Yes, and it would not end on a `ZeroDivisionError`, it is safe.

File: httpie/downloads.py (hunk 1 above)
Comment (BoboTiG) [reply httpie/downloads.py:25]: To ease monkey patching or eventual future change. I will revert, it is not really necessary.

**Comment by BoboTiG** [comment]
The title is misleading, you are right. I'll reword it.
